lib/pure/os.nim

```diff
@@ -3217,6 +3184,48 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; bufferSize = 8192.Positive): bool {.
+    rtl, extern: "nos$1", tags: [ReadIOEffect], noWeirdTarget.} =
+  ## Returns true if both pathname arguments refer to files with identical
+  ## binary content.
+  ##
+  ## If `checkSize` is `true` then checks the file sizes *before reading the files*,
+  ## if the files have different file sizes they can not have the same contents,
+  ## `checkSize = true` may be faster, specially for very big files.
+  ##
+  ## See also:
+  ## * `sameFile proc <#sameFile,string,string>`_
+  var
+    a, b: File
+  if not open(a, path1): return false
```

Comment:
* this should raise when you can't `open` (as in before this PR) instead of silently ignoring errors
* ditto with path2, taking care of correctness wrt fd leaks